src/txmempool.cpp

```diff
@@ -831,11 +832,17 @@ TxMempoolInfo CTxMemPool::info(const GenTxid& gtxid) const
 
 TxMempoolInfo CTxMemPool::info(const uint256& txid) const { return info(GenTxid{false, txid}); }
 
-void CTxMemPool::PrioritiseTransaction(const uint256& hash, const CAmount& nFeeDelta)
+bool CTxMemPool::PrioritiseTransaction(const uint256& hash, const CAmount& nFeeDelta)
 {
+    if (nFeeDelta == std::numeric_limits<CAmount>::min()) {
+        return false;
+    }
```

Comment:
Why?

Reply:
Otherwise the following invalid integer negation will take place:

```
util/moneystr.cpp:16:34: runtime error: negation of -9223372036854775808 cannot be represented in type 'CAmount' (aka 'long'); cast to an unsigned type to negate this value to itself
    #0 0x55b386a14eac in FormatMoney[abi:cxx11](long const&) src/util/moneystr.cpp:16:34
    #1 0x55b3861962f3 in CTxMemPool::PrioritiseTransaction(uint256 const&, long const&) src/txmempool.cpp:861:77
```

`FormatMoney(CAmount n)` is not defined for `n == std::numeric_limits<CAmount>::min()`.

The negation of `std::numeric_limits<CAmount>::min()` (-9223372036854775808) cannot be represented in the type `CAmount (int64_t)`.

Reply:
If the fix in #20406 is applied we don't have to have the restriction at this call site. I think that would be even better :)